Homework/Homework Remedial/Week 3.py

```diff
@@ -0,0 +1,129 @@
+    else:
+        return find_nth_number(n - 1) + 7
+
+
+#
+# Question 3
+
+base_url = "https://codefirstgirls.com/"
+current_url = base_url
+
+while True:
+    print("You are currently on the URL", current_url)
+    print("Where are you clicking?")
+
+    if current_url == base_url:
+        print("Options: Courses, Opportunities")
+        choice = input()
+        if choice.lower() == "courses":
+            current_url += "courses"
+        elif choice.lower() == "opportunities":
+            current_url += "opportunities"
+
+    elif current_url == base_url + "courses":
+        print("Options: CFGDegree, Back")
+        choice = input()
+        if choice.lower() == "cfgdegree":
+            current_url += "/cfgdegree"
+        elif choice.lower() == "back":
+            current_url = base_url
+
+    elif current_url == base_url + "opportunities":
+        print("Options: Ambassadors, Back")
+        choice = input()
+        if choice.lower() == "ambassadors":
+            current_url += "/ambassadors"
+        elif choice.lower() == "back":
+            current_url = base_url
+
+    elif current_url == base_url + "courses/cfgdegree":
+        print("Options: Back")
+        choice = input()
+        if choice.lower() == "back":
+            current_url = base_url + "courses"
+
+    elif current_url == base_url + "opportunities/ambassadors":
+        print("Options: Back")
+        choice = input()
+        if choice.lower() == "back":
+            current_url = base_url + "opportunities"
+
+    print()
+
+
+# Question 3b
+
+# The time complexity of the solution can be considered as O(1) since the number of URLs and user
+# choices does not have a significant impact on the number of iterations or operations performed by
+# the program. The number of URLs is fixed, and the user choices only affect the navigation path within
+# the predetermined URLs.
+#
+# The space complexity of the solution is also O(1) since the amount of memory used by the program
+# remains constant regardless of the number of URLs or user choices. The program only stores the base
+# URL and the current URL, and it does not require additional data structures or memory allocation that grows with the input size.
+#
+# Assumptions:
+#
+# Firstly, the provided URLs are the only valid URLs in the simulation. Other URLs or variations are not
+# considered.
+#
+# Secondly, the user's input is assumed to be valid and matches the available options exactly. ' \'No input validation or error handling is implemented.
+#
+# Thirdly, the simulation loop runs indefinitely until an external force or condition breaks the loop.
+#
+# Fourthly, the program does not handle cases where the user navigates to an invalid or non-existent URL.
+# Error handling and validation are not included in the code.
+#
+# Finally, the program focuses on simulating the navigation and printing the current URL after each move,
+# rather than implementing full website functionality or page rendering.
```

Comment:
Homework 3, Question 3 marks: 9/15.

Well done. 
However, one good improvement would have been to use a dictionary to link all 5 urls/pages to the choices available, and a list (or an equivalent structure) to track the current URL, instead of hardcoding them in the if/else statement.